frontend/src/components/SettingsConnections.vue

```diff
@@ -0,0 +1,151 @@
+      :message="t('text.settings.connectedAccounts.disconnect.zoom.message')"
+      :confirm-label="t('text.settings.connectedAccounts.disconnect.zoom.confirm')"
+      :cancel-label="t('text.settings.connectedAccounts.disconnect.zoom.cancel')"
+      :use-caution-button="true"
+      @confirm="() => disconnectAccount('zoom')"
+      @close="closeModals"
+  ></ConfirmationModal>
+</template>
+
+<script setup>
+import {
+  ref, inject, onMounted, computed,
+} from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
+import { useUserStore } from '@/stores/user-store';
+import { storeToRefs } from 'pinia';
+import CautionButton from '@/elements/CautionButton.vue';
+import ConfirmationModal from '@/components/ConfirmationModal.vue';
+import PrimaryButton from '@/elements/PrimaryButton.vue';
+import SecondaryButton from '@/elements/SecondaryButton.vue';
+import TextButton from '@/elements/TextButton.vue';
+
+// icons
+import { IconExternalLink } from '@tabler/icons-vue';
+
+// stores
+import { useExternalConnectionsStore } from '@/stores/external-connections-store';
+import { useCalendarStore } from '@/stores/calendar-store';
+
+// component constants
+const { t } = useI18n({ useScope: 'global' });
+const call = inject('call');
+const router = useRouter();
+const externalConnectionsStore = useExternalConnectionsStore();
+const calendarStore = useCalendarStore();
+const { connections } = storeToRefs(externalConnectionsStore);
+const { $reset: resetConnections } = externalConnectionsStore;
+
+const fxaEditProfileUrl = inject('fxaEditProfileUrl');
+
+const disconnectZoomModalOpen = ref(false);
+const disconnectGoogleModalOpen = ref(false);
+
+const closeModals = () => {
+  disconnectZoomModalOpen.value = false;
+  disconnectGoogleModalOpen.value = false;
+};
+
+const refreshData = async () => {
+  // Need to reset calendar store first!
+  await calendarStore.$reset();
+  await Promise.all([
+    externalConnectionsStore.fetch(call),
+    calendarStore.fetch(call),
+  ]);
+};
+
+onMounted(async () => {
+  await refreshData();
+});
+
+const displayModal = async (category) => {
+  if (category === 'zoom') {
+    disconnectZoomModalOpen.value = true;
+  } else if (category === 'google') {
+    disconnectGoogleModalOpen.value = true;
+  }
+};
+
+const connectAccount = async (category) => {
+  if (category === 'zoom') {
+    const { data } = await call('zoom/auth').get().json();
+    // Ship them to the auth link
+    window.location.href = data.value.url;
+  } else if (category === 'google') {
+    await router.push('/settings/calendar');
+  }
+};
+
+const disconnectAccount = async (category) => {
+  if (category === 'zoom') {
+    await call('zoom/disconnect').post();
+  } else if (category === 'google') {
+    await call('google/disconnect').post();
+  }
+
+  await resetConnections();
+  await refreshData();
+  await closeModals();
+};
+
+const editProfile = async () => {
+  window.location = fxaEditProfileUrl;
```

Comment:
Same as the profile page's edit profile button.

Reply:
suggestion (non-blocking): Can we do a router push here too? To preserve some history? Untested:

```suggestion
  window.location = router.push({ path: fxaEditProfileUrl });
```

Reply:
It's an external url so we cannot.

Reply:
Oooohh haha 😅 My bad. Totally read that wrong 😁

Reply:
np!